README.md

```diff
@@ -4,563 +4,133 @@
-
-## Initial state
-In order to specify the initial state, the empty argument constructor is used. It is guaranteed that before every test invocation a new test class instance is created.
-
-## Operations and groups
-As described above, each operation is specified via `@Operation` annotation.
-
-```java
-@Operation
-public Integer poll() { return q.poll(); }
-```
-
-### Calling at most once
-If an operation should be called at most once during the test execution, you can set `@Operation(runOnce = true)` option and this operation appears at most one time in the generated scenario.
-
-### Exception as a result
-If an operation can throw an exception and this is a normal result (e.g. `remove` method in `Queue` implementation throws `NoSuchElementException` if the queue is empty), it can be handled as a result if `@Operation(handleExceptionsAsResult = ...)` options are specified. See the example below where `NoSuchElementException` is processed as a normal result.
-
-```java
-@Operation(handleExceptionsAsResult = NoSuchElementException.class)
-public int remove() { return queue.remove(); }
-```
-
-### Operation groups
-In order to support single producer/consumer patterns and similar ones, each operation could be included in an operation group. Then the operation group could have some restrictions, such as non-parallel execution.
 
-In order to specify an operation group, `@OpGroupConfig` annotation should be added to the test class with the specified group name and its configuration:
+Lincheck is a practical and user-friendly framework for testing concurrent algorithms on the JVM. It provides a simple
+and declarative way to write concurrent tests.
 
-* **nonParallel** - if set all operations from this group will be invoked from one thread.
+With the Lincheck framework, instead of describing how to perform tests, you can specify _what to test_
+by just declaring all the operations to examine. After that, Lincheck automatically 
+generates a set of random concurrent scenarios,
+examines them using either stress-testing or bounded model checking, and
+verifies that the results of each invocation satisfy the required correctness property (linearizability by default).
 
-Here is an example with single-producer multiple-consumer queue test:
+## Documentation and Presentations
 
-```java
-@OpGroupConfig(name = "producer", nonParallel = true)
-public class SPMCQueueTest {
-  private SPMCQueue<Integer> q = new SPMCQueue<>();
-  
-  @Operation(group = "producer")
-  public void offer(Integer x) { q.offer(x); }
-  
-  @Operation
-  public Integer poll() { return q.poll(); }
-}
-```
-
-A generator for `x` parameter is omitted and the default one is used. See [Parameter generators](#parameter-generators) paragraph for details.
+Please see the [official tutorial](/docs/topics/lincheck-guide.md) that shows how to use Lincheck in various scenarios.
```

Comment:
that explains various features of Lincheck with a series of examples?

Reply:
"that presents main Lincheck features through examples"?